README.md

```diff
@@ -1,33 +1,40 @@
 # SimpleIndexer
 
-`SimpleIndexer` is a [mmap](https://docs.python.org/3/library/mmap.html)-based Indexer. 
-`SimpleIndexer` is recommended to be used when you have less than one million Documents. 
+`SimpleIndexer` use `DocumentArrayMemmap` for indexing `Document`. It is recommended to be used in most of the simple use cases when you have less than one million `Document`. 
 
+`SimpleIndexer` leverages `DocumentArrayMmap`'s [`match`](https://docs.jina.ai/api/jina.types.arrays.abstract/#jina.types.arrays.abstract.AbstractDocumentArray.match) function and searches the `k` nearst neighbors for the query `Document` based on their `embedding` field by the brutal-force search. By default, it calculates the `cosine` distance and returns all the indexed `Document`.
 
-It stores the [Document](https://docs.jina.ai/fundamentals/document/document-api/)s at the directory specified by `workspace`, 
-which can be defined using the `uses_metas` argument as such:
 
-```python
-Flow().add(
-    uses=SimpleIndexer,
-    uses_metas={'workspace': './foo_workspace'})
-```
 
-To search documents, the `SimpleIndexer` leverages [`DocumentArrayMmap`](https://docs.jina.ai/fundamentals/document/documentarraymemmap-api/)'s `match` function. 
-For advanced usages, one can pass arguments to the `match` function by specifying `match_args` as below:
+## Advanced Usages
+
+### Configure the index directory
+
+`SimpleIndexer` stores the `Document` at the directory, which is specified by `workspace` field under the [`metas`](https://docs.jina.ai/fundamentals/executor/executor-built-in-features/#meta-attributes) attribute. 
+
+You can find how to override `metas` attributes at [docs.jina.ai](https://docs.jina.ai/fundamentals/flow/add-exec-to-flow/#override-metas-configuration)
+
+
+### Configure the search behaviors
+
+You can use `match_args` argument to pass arguments to the `match` function as below. Please refer to [docs.jina.ai](https://docs.jina.ai/fundamentals/flow/add-exec-to-flow/#override-with-configuration) for more details about overriding `with` configurations. You can find more about the `match` function at [here](https://docs.jina.ai/api/jina.types.arrays.abstract/#jina.types.arrays.abstract.AbstractDocumentArray.match)
 
 ```python
 Flow().add(
     uses=SimpleIndexer,
     uses_with={
         'match_args': {
-            'metric': 'cosine',
-            'use_scipy': False}})
+            'metric': 'euclidean',
+            'use_scipy': True,
+            'limit': 10}})
```

Comment:
Since dynamic parameters/runtime args feature is recently added to `SimpleIndexer`, perhaps it would also be useful to a description about that?

````suggestion
```python
f =  Flow().add(
     uses=SimpleIndexer,
     uses_with={
         'match_args': {
             'metric': 'euclidean',
             'use_scipy': True,
             'limit': 10}})
 ```
Users can also pass runtime arguments to the `SimpleIndexer` at search time. This can be useful when users want to query with different match arguments. For instance, if users want to query results with a different top k or with a different metric, they can do so by defining a custom `limit` and `metric` in `parameters`:
```python
with f:
    f.search(inputs=Document(text='hello'), parameters={'limit': 5, 'metric': 'cosine'})
```
````